### PR TITLE
Implement support for read concerns

### PR DESCRIPTION
**Pull request by stefano-pogliani**
This PR addresses the read concern support part of #278.

**Note**: this PR introduces a breaking change because it requires the default read concern and max time ms attributes to be passed to some methods.

### CODE REVIEW

File: src/coll/mod.rs [hunk 1]
```diff
@@ -43,10 +45,13 @@ impl Collection {
         db: Database,
         name: &str,
         create: bool,
+        max_time_ms: Option<i64>,
```
Comment (saghm) [inline src/coll/mod.rs]: In retrospect,everything beyond the `Database` and Name should probably be abstracted to an options struct (and ditto for creating a database only needing `Client` and name). Are you interested in taking a stab at this, or would you prefer that I make the change first and then you can rebase it?

File: src/common.rs [hunk 2]
```diff
@@ -6,6 +6,62 @@ use bson::{self, Bson};
 use std::collections::BTreeMap;
 use std::str::FromStr;
 
+/// Level of desired consistency and isolation properties of the data read.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
```
Comment (saghm) [inline src/common.rs]: Out of curiosity, what's the rationale for deriving ordering on `ReadConcern`? I don't really think of read concerns has having a natural order, and I'm not sure that anyone would be able to intuitively guess the ordering of a given pair of read concerns.

File: script/start_mongo_release [hunk 3]
```diff
@@ -16,9 +16,17 @@ if [ ! -d "$install_dir/$release" ]; then
 fi
 
 # Start mongod processes
+enable_read_concern=""
+if [[ "$release" == *"3.2."* ]]; then
+  enable_read_concern="--enableMajorityReadConcern"
```
Comment (saghm) [inline script/start_mongo_release]: Is there a reason we need the tests to run with majority read concern?

**Review by saghm — COMMENTED** [review]
Thanks for the PR, and apologies for taking so long to review this! If you're still interested in working on this, I have a couple of questions and suggestions. If not, I totally understand, and I can make the additional changes and get it merged.

**Comment by stefano-pogliani** [comment]
Hi @saghm and thank you for the feedback.
I'll be happy to look at your comments but I don't think I'll have the time this week.

**Comment by saghm** [comment]
Hello everyone! We've just released an alpha of [an official MongoDB Rust driver](https://github.com/mongodb/mongo-rust-driver); you can read about it [here](https://www.mongodb.com/blog/post/announcing-the-official-mongodb-rust-driver)! Because of this, we won't be updating this driver anymore, and the repository will be made read-only as soon as I get finished posting these messages. Thanks for using this driver, and we hope you try out the new one!

**Pull request closed without merging**